homeassistant/components/sector_alarm/alarm_control_panel.py

```diff
@@ -0,0 +1,86 @@
+"""The Sector Alarm Integration."""
+import logging
+from homeassistant.components.sector_alarm import DOMAIN as SECTOR_DOMAIN
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.const import (
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_PENDING,
+)
+
+import homeassistant.components.sector_alarm as sector_alarm
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Initialition of the platform."""
+    sector_connection = hass.data.get(SECTOR_DOMAIN)
+
+    if sector_connection is None:
```

Comment:
The platform setup should only start if the component is ready and the connection established.

Reply:
Can i get some tips here? I thought that would be handled in __init__ by:
    except RuntimeError:
        _LOGGER.error("Could not login. Wrong username or password?")
        return